homeassistant/components/home_connect/switch.py

```diff
@@ -303,6 +304,17 @@ async def async_turn_on(self, **kwargs: Any) -> None:
 
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Switch the device off."""
+        if not hasattr(self, "power_off_state"):
+            await self.async_fetch_power_off_state()
+            if not hasattr(self, "power_off_state"):
```

Comment:
So we already fetch this property at boot, how can it still be different at runtime?

Reply:
During boot, there could be an issue with the communication that prevents fetching, so it might be necessary to attempt fetching again later

Reply:
How flaky is home connect? If you experience the service as flakey I would think we should make sure we retry the call, otherwise I would just say, fuck it we ball and see if it causes any problems.

Reply:
I never had any problem at boot, so let's do that.